### PR TITLE
Update client validation error messages to align with Heuristic evaluation criteria

### PR DESCRIPTION
**Pull request by diondrapeck**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

criteria: https://hits.microsoft.com/collection/7002637/Section/72548#5YP3J-design%22

- Make all text white except error message which should be red
- Precede error messages with "(x)"
- Make all links cyan

image

This PR also updates SDK validation error messages to raise MlExceptions and record error type when telemetry is recorded.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/_exception_helper.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
     SCHEMA_VALIDATION_ERROR_TEMPLATE,
     YAML_CREATION_ERROR_DESCRIPTION,
 )
-from azure.ai.ml.exceptions import ErrorTarget, ValidationErrorType, ValidationException
+from azure.ai.ml.exceptions import ErrorTarget, ValidationErrorType, ValidationException, MlException
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_exception_helper.py:18]: Does the lower case "l" bother you as much as it does to me?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_exception_helper.py [hunk 2]
```diff
@@ -86,7 +86,7 @@ def format_details_section(
 
     if hasattr(error, "message"):
         error_types[error.error_type] = True
-        details += f"\n\n{error.message}\n"
+        details += f"\n\n{Fore.RED}{error.message}{Fore.RESET}\n"
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_exception_helper.py:89]: This is very cool. How does it look like in a jupyter notebook?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_exception_helper.py (hunk 1 above)
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_exception_helper.py:18]: haha it does. If we don't have a reason for it to be lowercase, I can make it MLException

**Review by nancy-mejia — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/_exception_helper.py (hunk 2 above)
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_exception_helper.py:89]: image

Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_exception_helper.py:89]: image

**Pull request merged**